### PR TITLE
Turn dj-starter into a cookiecutter template.

### PR DESCRIPTION
**Pull request by DropsOfSerenity**
This changes dj-starter into a cookiecutter template. See README.md for details as it's been completely rewritten.

To test:

Follow the instructions in README.md and see if you can spin up a new project based on the instructions there. The only thing you will need to change (while this isn't merged in) is this command: `cookiecutter gh:Shift3/dj-starter --checkout feature/cookiecutter`

Other Changes:
* CircleCI changed to fit new cookiecutter template. CircleCI now generates the project and runs the tests to ensure everything works on the generated project.
* Proprietary `LICENSE` used in generated project.
* Project `README` moved to generated project, rewrote `README` to describe how to start a project with cookiecutter.

### CODE REVIEW

File: cookiecutter.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+	"project_name": "My Project",
+	"project_description": "A description of what the project is",
+	"project_slug": "{{ cookiecutter.project_name.lower()|replace(' ', '_')|replace('-', '_')|replace('.', '_')|trim() }}",
+	"client_name": "The client's company name",
+	"staging_client_url": "https://my-react-client.shift3sandbox.com"
+}
```
Comment (AramayisO) [inline cookiecutter.json:7]: Nice! I really like how simple this config file is :fire:

File: {{ cookiecutter.project_slug }}/README.md [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+# From within docker container
+$ pytest
+```
+
+## The Batteries
+
+The starter project is **Batteries Included**, meaning it comes with lots of helpful feature preinstalled for you, to make your life as a developer easier. What follows is a list of what we include and where to find more information about them.
+
+* [Django](https://www.djangoproject.com/) is the framework that we are building on top of. If you are unfamiliar with django, it has a great [tutorial](https://www.djangoproject.com/start/) to get you introduced to the basic concepts.
+* [Django REST Framework](https://www.django-rest-framework.org/) (also known as DRF) specializes django to excel in serving a great API. It provides, serialization, authentication, and a browseable API that you can use in development to aid your API development experience.
```
Comment (joshwingreene) [inline {{ cookiecutter.project_slug }}/README.md]: Slight typo here. I would remove the comma in between 'provides' and 'serialization'.

**Review by joshwingreene — APPROVED** [review]
I was able to generate the project successfully.

Besides that, I found a slight typo in the README.

Nice work!

**Review by AramayisO — APPROVED** [review]
This is really awesome :rocket:  Tested and works as expected.

The only thing I felt missing from the README of the generated project was a quickstart of how to setup and run locally without Docker. But I'm not sure, maybe we're not supporting local development anymore? Either way, I didn't have too much trouble, just followed the error messages to set up the expected env vars and such.

File: {{ cookiecutter.project_slug }}/README.md (hunk 2 above)
Comment (DropsOfSerenity) [reply {{ cookiecutter.project_slug }}/README.md]: Fixed, thanks :)

**Pull request merged**